crates/macro-support/src/parser.rs

```diff
@@ -377,11 +377,17 @@ impl<'a> ConvertToAst<BindgenAttrs> for &'a mut syn::ItemStruct {
             .unwrap_or(self.ident.to_string());
         let is_inspectable = attrs.inspectable().is_some();
         let getter_with_clone = attrs.getter_with_clone().is_some();
+        let transparent = self.attrs.iter().any(|attr| {
+            matches!(attr.style, AttrStyle::Outer)
+                && attr.path.is_ident("repr")
+                && attr.tokens.to_string() == "(transparent)"
+        });
         for (i, field) in self.fields.iter_mut().enumerate() {
-            match field.vis {
-                syn::Visibility::Public(..) => {}
-                _ => continue,
+            let public = matches!(field.vis, syn::Visibility::Public(..));
+            if !transparent && !public {
```

Comment:
Can you add a comment for why `transparent` is factored in here?

Reply:
Yeah, it might not stay like this, because I'm not sure if this is indeed the best place to handle the `transparent` case. [I mentioned it in `#wg-wasm` on Discord](https://discord.com/channels/442252698964721669/443151097398296587/925143307368157205):

> [...] im not sure if i should just add a `transparent: bool` field to `ast::Struct` or make a new type to represent anything that's `#[repr(transparent)]`
> the implementation of `ToTokens` for `ast::Struct` would be quite different for transparent types, im not sure if branching there is a good idea